include/ipc/ipc_service.h

```diff
@@ -170,6 +170,80 @@ int ipc_service_register_endpoint(const struct device *instance,
  */
 int ipc_service_send(struct ipc_ept *ept, const void *data, size_t len);
 
+/** @brief Get an empty TX buffer to be sent using @ref ipc_service_send_nocopy
+ *
+ *  This function can be called to get an empty TX buffer so that the
+ *  application can directly put its data into the sending buffer without copy
+ *  from an application buffer.
+ *
+ *  It is the application responsibility to correctly fill the allocated TX
+ *  buffer with data and passing correct parameters to @ref
+ *  ipc_service_send_nocopy function to perform data no-copy-send mechanism.
+ *
+ *  @param ept Registered endpoint by @ref ipc_service_register_endpoint.
+ *  @param data Pointer to the empty TX buffer.
+ *  @param len Pointer to store the TX buffer size.
+ *  @param wait Boolean, wait or not for buffer to become available
+ *
+ *  @retval -EIO when no backend is registered or send hook is missing from
+ *	    backend.
+ *  @retval -EINVAL when instance or endpoint is invalid.
+ *  @retval -ENOTSUP when this is not supported by backend.
+ *  @retval -ENOBUFS when there are no TX buffers available.
```

Comment:
what about `-EALREADY` when buffer cannot be granted because one is already claimed. That may be the case for implementations which are using circular buffer like #43544.

Reply:
uhm, in theory you can keep requesting buffers until there are available buffers, so you are not forced to claim one single buffer at time. When there are no more available buffers you get `-ENOBUFS` so I don't think that having `-EALREADY` is useful or needed.

Reply:
in case of #43544 it will not support multiple `get_buffer` calls, `-ENOBUFS` would indicate that circular buffer is full and `-EALREADY` would indicate that `get_buffer` was called and it is not yet sent. In case of backend that supports multiple `get_buffers` `-EALREADY` would never be returned.

Reply:
ok makes sense